Amplify/Categories/Auth/Models/AuthSignInStep.swift

```diff
@@ -5,31 +5,57 @@
 // SPDX-License-Identifier: Apache-2.0
 //
 
+public typealias AdditionalInfo = [String: String]
+
 /// Auth SignIn flow steps
-public enum AuthSignInStep {
+///
+///
+public enum AuthSignInStep: Equatable {
 
     /// Auth step is SMS multi factor authentication.
     ///
     /// Confirmation code for the MFA will be send to the provided SMS.
-    case confirmSignInWithSMSMFACode
+    case confirmSignInWithSMSMFACode(AuthCodeDeliveryDetails, AdditionalInfo?)
 
     /// Auth step is in a custom challenge depending on the plugin.
     ///
-    case confirmSignInWithCustomChallenge
+    case confirmSignInWithCustomChallenge(AdditionalInfo?)
 
     /// Auth step required the user to give a new password.
     ///
-    case confirmSignInWithNewPassword
+    case confirmSignInWithNewPassword(AdditionalInfo?)
 
     /// Auth step required the user to change their password.
     ///
-    case resetPassword
+    case resetPassword(AdditionalInfo?)
 
     /// Auth step that required the user to be confirmed
     ///
-    case confirmSignUp
+    case confirmSignUp(AdditionalInfo?)
 
     /// There is no next step and the signIn flow is complete
     ///
     case done
 }
+
+// swiftlint:disable empty_enum_arguments
+extension AuthSignInStep {
```

Comment:
I can't remember whether Swift syntax allows us to declare a protocol conformance here instead of up on the type itself, but if so, let's declare `Equatable` conformance here...

...but...

Do we want equatable conformance? What is the use case we're solving with it? It's mildly surprising that two different MFA codes would be considered equal, even though the step itself might be the same. If statements can still be written in the somewhat cumbersome Swift style: `if case .confirmSignInWithSMSMFACode = someStateToInspect ... `

Reply:
Removed